app/validation/validators.py

```diff
@@ -167,3 +150,47 @@ def __call__(self, form, from_field, to_field):
             raise validators.ValidationError(self.messages['INVALID_DATE_RANGE_TO_FROM_SAME'])
         elif from_date > to_date:
             raise validators.ValidationError(self.messages['INVALID_DATE_RANGE_TO_BEFORE_FROM'])
+
+
+class FormResponseRequired(object):
+    field_flags = ('required', )
+
+    def __init__(self, message=None):
+        if not message:
+            message = error_messages['MANDATORY']
+        self.message = message
+
+    def __call__(self, form, field):
+
+        if hasattr(form, 'day'):
+            if not form.day.data and not form.month.data and not form.year.data:
+                raise validators.StopValidation(self.message)
+        elif hasattr(form, 'month'):
+            if not form.month.data and not form.year.data:
+                raise validators.StopValidation(self.message)
+        elif hasattr(form, 'hours') and hasattr(form, 'mins'):
+            if not form.hours.data and not form.mins.data:
+                raise validators.StopValidation(self.message)
+
+
+class TimeInputCheck(object):
```

Comment:
I agree with this point. I don't see what TimeInputCheck is giving us over the other validators